bool open 
---
uint8 result